### DISCUSSION TITLE
get config like README

### DISCUSSION THREAD

**Discussion by OmiChrome** [shellRaining/hlchunk.nvim#32] [Q&A] [answered]
this one ghost-indent
![image](https://github.com/shellRaining/hlchunk.nvim/assets/108952834/4c7aa031-1443-40ec-a0f1-65533d29f152)
i saw the same on Nvchad , but wan't able to replicate that on here

**Comment by OmiChrome** [comment]
also blank like this one  , i hate it when DOTS make confusing when u actually put .. dots on commets
![image](https://github.com/shellRaining/hlchunk.nvim/assets/108952834/25f11173-ba52-4200-baa2-14bd822e5f25)

**Reply by shellRaining** [reply, marked as answer]
you can get the detail config info by click the picture in README, you will jump to another docs about this mod, but maybe it is outdated, I will fix this when have time~

if you want to config like the first picture, you can write like this

```lua
require("hlchunk").setup({
  indent = {
      enable = true,
      use_treesitter = false,
      chars = {
          " ",
      },
      style = {
          { bg = vim.fn.synIDattr(vim.fn.synIDtrans(vim.fn.hlID("Whitespace")), "fg", "gui") },
      },
      exclude_filetype = exclude_ft,
  }
})
```

the second like this

```lua
blank = {
    enable = true,
    chars = {
        " ",
    },
    style = {
        { bg = vim.fn.synIDattr(vim.fn.synIDtrans(vim.fn.hlID("cursorline")), "bg", "gui") },
        { bg = "", fg = "" },
    },
}
```

You can choose your favorite color and style by yourself, but I highly recommend using colors with low saturation.

**Comment by OmiChrome** [comment]
why my CHUNK is hideen in some part? also chunk,blank is mostly buggy  they cover some words as mosty commets
![image](https://github.com/shellRaining/hlchunk.nvim/assets/108952834/aef2fa09-f982-47b4-878e-7e08747dbf04)

my theme is [rose-pine](https://github.com/rose-pine/neovim)

**Reply by OmiChrome** [reply]
![image](https://github.com/shellRaining/hlchunk.nvim/assets/108952834/767fb7f6-c11b-400e-a5bd-8db02ab59915)
this looks even MORE ugly,  
can u just introduce `OPACITY` ? i dont have much experience workign with colors so im still confused how to get like yours

**Reply by shellRaining** [reply]
The first issue you mentioned is that the chunk and blank are covered by comments during rendering. I think this is not a problem because if the characters are covered by the virtual text of the chunk or blank, it will cause misunderstanding.
The correct approach, I think, should be to format your code, as shown below.

![image](https://github.com/shellRaining/hlchunk.nvim/assets/111564053/f0daa303-2440-44dd-a62f-fff08380799d)

As for the second question, it can be alleviated by setting other colors. Regarding `opacity`, I haven't seen a similar option in Neovim yet. I remember that it can be achieved by setting the `blend` option. But please give me some time to learn. I will notify you again once there is progress.

The proposed modifications are as follows:

```lua
style = {
    { bg = vim.fn.synIDattr(vim.fn.synIDtrans(vim.fn.hlID("cursorline")), "bg", "gui") },
},
```

Another issue could be with your Neovim filetype settings. When opening a Lua file, your `shiftwidth` is set to 2. You can try typing the command `set shiftwidth=4` to resolve this.
